draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -310,26 +301,115 @@ authorized(AS x, AS y) =  / Else, "Provider+" if the VAP-SPAS entry
         If an attacker creates a route leak intentionally, they may try to strip their AS from the AS_PATH.
         To partly guard against that, a check is necessary to match the most recently added AS in the AS_PATH to the BGP neighbor's ASN.
         This check MUST be performed as specified in Section 6.3 of <xref target="RFC4271"/>.
-        If the check fails, then the AS_PATH is considered a Malformed AS_PATH and the UPDATE is considered to be in error (Section 6.3 of <xref target="RFC4271"/>).
+        If the check fails, then the AS_PATH is considered a Malformed AS_PATH and the UPDATE is in error (Section 6.3 of <xref target="RFC4271"/>).
         The case of transparent RS MUST also be appropriately taken care of (e.g., by suspending the neighbor ASN check).
-        The check fails also when the AS_PATH is empty (zero length) and such UPDATEs will also be considered to be in error.
+        The AS_PATH check fails also when it is empty (zero length) since such an UPDATE in eBGP is considered to be an error.
       </t>
       <t>
         <xref target="I-D.ietf-idr-deprecate-as-set-confed-set"/> specifies that "treat-as-withdraw" error handling <xref target="RFC7606"/> SHOULD be applied to routes with AS_SET in the AS_PATH.
         In the current document, routes with AS_SET are given Invalid evaluation in the AS_PATH verification procedures (<xref target="Upflow"/> and <xref target="Downflow"/>).
         See <xref target="mitigation"/> for how routes with Invalid AS_PATH are handled.
       </t>
+
+    <section title="Semantics of Valid, Invalid, and Unknown" anchor="semantics">
```

Comment:
I don't agree with this section restructure. It shoud start with definition of the up/down ramps and what is valid and invalid. The 'Unknown' state just grows fron the limitations of ASPA, that why with ASPA we can only measure min and max lengths, but not exact length of the ramps.

Reply:
@mitradir @cjeker 
Alexander: Sec. 7.1 provides high level semantics about (not how to determine) Invalid, Valid, and Unknown. I think Section 7.1 can be dropped in light of Sec. 7.2. Because Sec. 7.2 systematically covers what makes a route Invalid, Valid, or Unknown and also explains how they are determined using ASPA -- for upstream and downstream paths.

The basic definition of Invalid (route leak) is:  
   "For the AS path to be Invalid (route leak), there must be two hops 
    -- one on the left and another on the right -- facing each other 
    that are not C2P (i.e., they are P2C or P2P)."

Invalid is easily explained with the help of Fig. 2, not the up-ramp/down-ramp figure (Fig. 3).  Then Sec. 7.2.1 proceeds to show how ASPAs are used to determine Invalid.

The up-ramp/down-ramp figure (Fig. 3 in Sec. 7.2.2) is used to explain Valid and Unknown for downstream paths. The apexes (lengths) of up-ramp and down-ramp are *exactly* determined by ASPAs. It is not correct to say that there are min and max associated with them. The up-ramp and down-ramp are determined by contiguous hops that are Provider+ (i.e., C2P) based on ASPA. Fig. 3 is primarily useful for finding Valid or Unknown only for the downstream paths.

Sec. 7.3 and 7.4 are essentially the same as what you currently have in the Master. No? 

I would like to request you to read Sec. 7.2, 7.3, and 7.4 and offer specific comments on it. I am happy to do a Zoom call and discuss any concerns/questions you may have.